python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+                for i,ix in enumerate(indices):
+                    magList[ix] = magArray[i]
+            else:
+                magList = dummySed.manyMagCalc(self._phiArray, self._wavelenStep)
+
+            return numpy.array(magList)
+
+        else:
+            return numpy.array([numpy.NaN]*self._nBandpasses)
+
+
+
+    def calcMagListFromSedList(self, sedList, indices=None):
+        """
+        Return a 2-D array of magnitudes from a SedList.
+        Each row will correspond to a different Sed, each column
+        will correspond to a different bandpass.
+
+        @param [in] sedList is a SedList containing the Seds
+        whose magnitudes are desired.
+
+        @param [in] indices is an optional list of indices indicating which bandpasses to actually
+        calculate magnitudes for.  Other magnitudes will be listed as 'None' (i.e. this method will
+        return as many magnitudes as were loaded with the loadBandpassesFromFiles methods; it will
+        just return nonsense for magnitudes you did not actually ask for)
+
+        @param [out] output_list is a 2-D numpy array containing the magnitudes
+        of each Sed (the rows) in each bandpass contained in this BandpassDict
+        (the columns)
+        """
+
+        one_at_a_time = False
+        if sedList.wavelenMatch is None:
+            one_at_a_time = True
+        elif len(sedList.wavelenMatch) != len(self._wavelen_match):
+            one_at_a_time = True
+        elif not numpy.allclose(sedList.wavelenMatch, self._wavelen_match, atol=0.0, rtol=1.0e-6):
+            one_at_a_time = True
+
+        output_list = []
+        if one_at_a_time:
+            for sed_obj in sedList:
+                sub_list = self.calcMagListFromSed(sed_obj, indices=indices)
+                output_list.append(sub_list)
+        else:
+            if indices is not None:
+                for sed_obj in sedList:
+                    sub_list = numpy.array([numpy.NaN]*self._nBandpasses)
+                    if sed_obj.wavelen is not None:
+                        sed_obj.flambdaTofnu()
+                        mag_list = sed_obj.manyMagCalc(self._phiArray, self._wavelenStep, observedBandpassInd=indices)
+                        for i,ix in enumerate(indices):
+                            sub_list[ix] = mag_list[i]
+                    output_list.append(sub_list)
+            else:
+                for sed_obj in sedList:
+                    if sed_obj.wavelen is None:
+                        sub_list = numpy.array([numpy.NaN]*self._nBandpasses)
+                    else:
+                        sed_obj.flambdaTofnu()
+                        sub_list = sed_obj.manyMagCalc(self._phiArray, self._wavelenStep)
+                    output_list.append(sub_list)
+
+        return numpy.array(output_list)
+
+
+    def calcFluxListFromSed(self, sedobj, indices=None):
+        """
+        Return a list of Fluxes for a single Sed object.
```

Comment:
I've tried calculating magnitues and fluxes without converting flambda to fnu.  It doesn't go well.  Any time anything in the Sed changes, fnu gets set to None to prevent flambda and fnu from getting out of sync.  It is always safer to calculate fnu right before integrating over a bandpass throughput.